hibernate-core/src/main/java/org/hibernate/dialect/OracleDialect.java

```diff
@@ -771,7 +771,14 @@ else if ( precision <= 53 ) {
 					// Don't infer TINYINT or SMALLINT on Oracle, since the
 					// range of values of a NUMBER(3,0) or NUMBER(5,0) just
 					// doesn't really match naturally.
-					if ( precision <= 10 ) {
+
+					// Backwards compatibility patch:
+					// In Oracle8iDialect (maybe also in previous and
+					// intermediate versions) all DECIMALs used BIGINT JDBC
+					// type descriptor.
+					boolean tryInteger = false;
+
+					if ( tryInteger && precision <= 10 ) {
```

Comment:
I don't understand why you would propose this change when the reason for using `INTEGER` is explained quite carefully and clearly in a comment on _the very next line of code_.

Reply:
Yes, this was an intentional change made in a new major version.

That's the whole purpose of having major releases. We would have called it Hibernate 5.7 if we were committed to strict backward compatibility.